File: src/ua/edu/odeku/ceem/mapRadar/tools/ToolFrame.java
package ua.edu.odeku.ceem.mapRadar.tools;

import ua.edu.odeku.ceem.mapRadar.utils.thread.Handler;

import javax.swing.*;
import java.awt.*;

/**
 * Общее окно для всех Tool компонентов программы
 *
 *
 * User: Aleo skype: aleo72
 * Date: 10.11.13
 * Time: 15:57
 */
public class ToolFrame extends JFrame {

    protected JPanel panel;
    protected Handler handler;

    public ToolFrame(CeemRadarTool ceemRadarTool, String title){
        this.panel = ceemRadarTool.getPanel();
        ceemRadarTool.setParent(this);
        handler = ceemRadarTool.getHandlerForJFrame(this);
        this.setTitle(title);
        this.setLocationByPlatform(true);
        initToolFrameSettings();
    }

    protected void initToolFrameSettings(){
        this.setAlwaysOnTop(true);
        this.setLayout(new BorderLayout());
        this.add(panel,BorderLayout.CENTER);

        this.pack();

        this.setMinimumSize(new Dimension(this.getWidth(), this.getHeight()));

        if(handler != null){
            handler.start();
        }
    }
}

File: src/ua/edu/odeku/ceem/mapRadar/tools/ToolFrame.scala
/*
 * Odessa State environmental University
 * Copyright (C) 2014
 */

package ua.edu.odeku.ceem.mapRadar.tools

import javax.swing.JFrame
import java.awt.{Dimension, BorderLayout}
import java.awt.event.{WindowEvent, WindowAdapter}

/**
 * Общее окно для всех Tool компонентов программы
 *
 * User: Aleo Bakalov
 * Date: 10.12.13
 * Time: 16:28
 */
class ToolFrame(val ceemRadarTool: CeemRadarTool, val titleToolFrame: String) extends JFrame() {

	def this(toolName: String, titleToolFrame: String){
		this(Class.forName(toolName).newInstance().asInstanceOf[CeemRadarTool] , titleToolFrame)
	}

	ceemRadarTool.setParent(this)

	protected val startFunction: (ToolFrame) => Unit = ceemRadarTool.startFunction
	protected val endFunction: (ToolFrame) => Unit = ceemRadarTool.endFunction

	this.setTitle(titleToolFrame)

	this.setLocationByPlatform(true)

	initToolFrameSettings()

	protected def initToolFrameSettings() {
		//		this.setAlwaysOnTop(true)
		this.setLayout(new BorderLayout)
		this.add(ceemRadarTool.rootPanel, BorderLayout.CENTER)
		this.pack()
		this.setMinimumSize(new Dimension(this.getWidth, this.getHeight))
		if (startFunction != null) {
			startFunction.apply(this)
		}
	}

	this.addWindowListener(new WindowAdapter {
		override def windowClosing(e: WindowEvent) {
			if (endFunction != null) {
				endFunction.apply(ToolFrame.this)
			}
		}
	})

}
